x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/percentile_ranks.tsx

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFormRow, EuiFieldNumberProps, EuiFieldNumber } from '@elastic/eui';
+import React, { useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { AggFunctionsMapping } from '@kbn/data-plugin/public';
+import { buildExpressionFunction } from '@kbn/expressions-plugin/public';
+import { OperationDefinition } from '.';
+import {
+  getFormatFromPreviousColumn,
+  getInvalidFieldMessage,
+  getSafeName,
+  isValidNumber,
+  getFilter,
+  isColumnOfType,
+  combineErrorMessages,
+} from './helpers';
+import { FieldBasedIndexPatternColumnMultipleValues } from './column_types';
+import { adjustTimeScaleLabelSuffix } from '../time_scale_utils';
+import { useDebouncedValue } from '../../../shared_components';
+import { getDisallowedPreviousShiftMessage } from '../../time_shift_utils';
+
+export interface PercentileRanksIndexPatternColumn
+  extends FieldBasedIndexPatternColumnMultipleValues {
+  operationType: 'percentile_ranks';
+}
+
+function ofName(name: string, value: number, timeShift: string | undefined) {
+  return adjustTimeScaleLabelSuffix(
+    i18n.translate('xpack.lens.indexPattern.percentileRanksOf', {
+      defaultMessage: 'Percentile rank ({value}) of {name}',
+      values: { name, value },
+    }),
+    undefined,
+    undefined,
+    undefined,
+    timeShift
+  );
+}
```

Comment:
The label is following the aggBased format
Percentile rank (value) of <field>
<img width="391" alt="image" src="https://user-images.githubusercontent.com/17003240/169516936-a76088a5-d640-455f-9719-89cd19ce7afb.png">

Reply:
Looks good to me!